src/libinjector/win_injector.c

```diff
@@ -672,7 +681,14 @@ static event_response_t mem_callback(drakvuf_t drakvuf, drakvuf_trap_info_t* inf
         regs.x86.rip);
 
     regs.x86.rip = injector->exec_func;
-    injector->status = STATUS_CREATE_OK;
+    if (INJECT_METHOD_WRITE_FILE == injector->method)
+    {
+        injector->status = STATUS_ALLOC_OK;
+    }
+    else
+    {
+        injector->status = STATUS_CREATE_OK;
+    }
```

Comment:
please annotate that it's `CREATE_FILE` or `SHELLEXEC`

Reply:
Done.